src/Microsoft.Health.Fhir.Shared.Api.UnitTests/Controllers/ConvertDataControllerTests.cs

```diff
@@ -60,40 +86,62 @@ public async Task GivenAConvertDataRequest_WhenInvalidBodySent_ThenRequestNotVal
             await Assert.ThrowsAsync<RequestNotValidException>(() => _convertDataeEnabledController.ConvertData(body));
         }
 
+        [Theory]
+        [MemberData(nameof(InconsistentBody), MemberType = typeof(ConvertDataControllerTests))]
+        public async Task GivenAConvertDataRequest_WhenInconsistentBodySent_ThenInconsistentThrown(Parameters body)
+        {
+            _convertDataeEnabledController.ControllerContext.HttpContext.Request.Method = HttpMethods.Post;
```

Comment:
>_convertDataeEnabledController.ControllerContext.HttpContext.Request.Method = HttpMethods.Post; [](start = 12, length = 95)

I don't understand this line. There is direct call to controller method, you don't have to set method type in  context to just call method. In e2e tests you do, because you invoke controller action via going through all the debris of ASP.NET, but to just call method on object...

I also don't see any checks inside controller (other than attribute) for method (and it's correct).

Can we remove this line from this test and other tests in this file.

Reply:
Thanks for pointing out it. I removed and tested it, it works! Now, all the controllerContext are removed in this file.